internal/runners/checkout/checkout.go

```diff
@@ -77,32 +77,35 @@ func NewCheckout(prime primeable) *Checkout {
 }
 
 func (u *Checkout) Run(params *Params) (rerr error) {
-	var ns project.Namespaced
+	var err error
+	var ns *project.Namespaced
 	var archive *checkout.Archive
 
-	if strings.HasSuffix(params.Namespace, checkout.ArchiveExt) {
-		var err error
+	switch {
+	// Checkout from archive
+	case strings.HasSuffix(params.Namespace, checkout.ArchiveExt):
 		archive, err = checkout.NewArchive(params.Namespace)
 		if err != nil {
 			return errs.Wrap(err, "Unable to read archive")
 		}
 		defer archive.Cleanup()
-		ns = *archive.Namespace
+		ns = archive.Namespace
 		params.Branch = archive.Branch
-	} else if err := ns.Set(params.Namespace); err != nil {
```

Comment:
`ns.Set()` exists to satisfy and work with the captain flag/argument mechanic. There's no point in using it in this context.

We should ideally have separated the types for "namespace" and "namespace flag". Don't wanna go that far for this PR though.